test/e2e_node/pod_resize_test.go

```diff
@@ -0,0 +1,1588 @@
+					Resources: &ContainerResources{CPUReq: "100m", CPULim: "200m", MemReq: "100Mi", MemLim: "200Mi"},
+					CPUPolicy: &doRestart,
+					MemPolicy: &doRestart,
+				},
+				{
+					Name:      "c2",
+					Resources: &ContainerResources{CPUReq: "200m", CPULim: "300m", MemReq: "200Mi", MemLim: "300Mi"},
+					CPUPolicy: &doRestart,
+					MemPolicy: &noRestart,
+				},
+				{
+					Name:      "c3",
+					Resources: &ContainerResources{CPUReq: "300m", CPULim: "400m", MemReq: "300Mi", MemLim: "400Mi"},
+					CPUPolicy: &noRestart,
+					MemPolicy: &doRestart,
+				},
+			},
+			patchString: `{"spec":{"containers":[
+						{"name":"c2", "resources":{"requests":{"cpu":"250m","memory":"250Mi"},"limits":{"cpu":"350m","memory":"350Mi"}}},
+						{"name":"c3", "resources":{"requests":{"cpu":"100m","memory":"100Mi"},"limits":{"cpu":"200m","memory":"200Mi"}}}
+					]}}`,
+			expected: []TestContainerInfo{
+				{
+					Name:      "c1",
+					Resources: &ContainerResources{CPUReq: "100m", CPULim: "200m", MemReq: "100Mi", MemLim: "200Mi"},
+					CPUPolicy: &doRestart,
+					MemPolicy: &doRestart,
+				},
+				{
+					Name:         "c2",
+					Resources:    &ContainerResources{CPUReq: "250m", CPULim: "350m", MemReq: "250Mi", MemLim: "350Mi"},
+					CPUPolicy:    &noRestart,
+					MemPolicy:    &noRestart,
+					RestartCount: 1,
+				},
+				{
+					Name:         "c3",
+					Resources:    &ContainerResources{CPUReq: "100m", CPULim: "200m", MemReq: "100Mi", MemLim: "200Mi"},
+					CPUPolicy:    &doRestart,
+					MemPolicy:    &doRestart,
+					RestartCount: 1,
+				},
+			},
+		},
+	}
+
+	for idx := range tests {
+		tc := tests[idx]
+		ginkgo.It(tc.name, func(ctx context.Context) {
+			cpuCap, _, _ = getLocalNodeCPUDetails(ctx, f)
+
+			// Skip InPlacePodVerticalScaling tests altogether if the CPU capacity < 2.
+			if cpuCap < 2 {
+				e2eskipper.Skipf("Skipping InPlacePodResize tests since the CPU capacity < 2")
+			}
+
+			var testPod *v1.Pod
+			var patchedPod *v1.Pod
+			var pErr error
+			tStamp := strconv.Itoa(time.Now().Nanosecond())
+			initDefaultResizePolicy(tc.containers)
+			initDefaultResizePolicy(tc.expected)
+			testPod = makeTestPod(f.Namespace.Name, "testpod", tStamp, tc.containers)
+
+			ginkgo.By("creating pod")
+			newPod := podClient.CreateSync(ctx, testPod)
+
+			ginkgo.By("verifying the pod is in kubernetes")
+			selector := labels.SelectorFromSet(labels.Set(map[string]string{"time": tStamp}))
+			options := metav1.ListOptions{LabelSelector: selector.String()}
+			podList, err := podClient.List(context.TODO(), options)
+			framework.ExpectNoError(err, "failed to query for pods")
+			gomega.Expect(podList.Items).Should(gomega.HaveLen(1))
+
+			ginkgo.By("verifying initial pod resources, allocations are as expected")
+			verifyPodResources(newPod, tc.containers)
+			ginkgo.By("verifying initial pod resize policy is as expected")
+			verifyPodResizePolicy(newPod, tc.containers)
+
+			ginkgo.By("verifying initial pod status resources")
```

Comment:
Probably want to wait for the pod to be running first.

Reply:
Done, using WaitForPodCondition